### PR TITLE
Adds  the other 3 40 milll rounds

### PR DESCRIPTION
**Pull request by thedarkcrusader**
Adds the other  40mill rouns smoke flash and white phos

## About The Pull Request
This PR adds the other 40mill grenade's flash and white phosphorus shells  Aswell as adjusts the armory to have 2 flash and 2 smokescreen boxes  flash and smokescreen can be bought from the Hellcat

## Why It's Good For The Game
Ive been needing to actually add in the others since i got those sprites made finallly got around to it this is allowing a lot more diverse use  of specialty equipment.  

## Testing
Spawned grenades and grenade boxes
fired  grenades
grenade's detonated Flash and white phos

smoke  fixed

## Changelog
:cl:
add: Flash shells, Smoke shells, White phosphorus 40 mm grenade shells and boxes,
add: The Hellcat can sell flash shells and smoke shells.
add:  The armory has 2 boxes of flash shells. 
/:cl:

### CODE REVIEW

**Comment by thedarkcrusader** [comment]
if you want to give me new Names please do

**Comment by Firefox13** [comment]
> smoke needs to be fixed

Do we need to convert the PR to draft, then?

**Comment by thedarkcrusader** [comment]
> > smoke needs to be fixed
> 
> Do we need to convert the PR to draft, then?

for the time being yeah i im going to try to fix it before its  time for the server to go up

File: code/modules/projectiles/ammunition/bullets.dm [hunk 1]
```diff
@@ -474,6 +474,41 @@
...
+	name = "SA WPR \"Gornja\""
+	desc = "A modernized incendiary fresh out of serbia very flammable, for clearing out ship's and villages."
+	icon_state = "warcrime"
+	projectile_type = /obj/item/projectile/bullet/grenade/handgrenade/white_phosphorous
+	w_class = ITEM_SIZE_SMALL
+	caliber = CAL_GRENADE
+	is_caseless = TRUE
+	maxamount = 1
+	spawn_tags = SPAWN_TAG_AMMO
+	rarity_value = 25
```
Comment (Firefox13) [inline code/modules/projectiles/ammunition/bullets.dm:511]:
```suggestion
	rarity_value = 25

```

File: code/modules/projectiles/projectile/projectilegrenades.dm [hunk 2]
```diff
@@ -104,3 +104,18 @@
 /obj/item/projectile/bullet/grenade/handgrenade/teargas    // Because why not
 	name = "cs shell"
 
+
+/obj/item/projectile/bullet/grenade/handgrenade/smoke
+	name = "smoke grenade"
+	hand_gren = /obj/item/grenade/smokebomb
+
+/obj/item/projectile/bullet/grenade/handgrenade/flash
+	name = "Flash grenade"
```
Comment (Firefox13) [inline code/modules/projectiles/projectile/projectilegrenades.dm:113]:
```suggestion
	name = "flash grenade"
```

File: code/modules/projectiles/projectile/projectilegrenades.dm [hunk 3]
```diff
@@ -104,3 +104,18 @@
...
+	hand_gren = /obj/item/grenade/flashbang
+
+
+/obj/item/projectile/bullet/grenade/handgrenade/white_phosphorous
+	name = "White phosphorus grenade"
```
Comment (Firefox13) [inline code/modules/projectiles/projectile/projectilegrenades.dm:118]:
```suggestion
	name = "white phosphorus grenade"
```
not like anyone is gonna read them but eh, they shouldn't be uppercases

File: code/modules/projectiles/ammunition/bullets.dm [hunk 4]
```diff
@@ -474,6 +474,41 @@
...
+	w_class = ITEM_SIZE_SMALL
+	caliber = CAL_GRENADE
+	is_caseless = TRUE
+	maxamount = 1
+	spawn_tags = SPAWN_TAG_AMMO
+	rarity_value = 15
+
+/obj/item/ammo_casing/grenade/white_phosphorous
+	name = "SA WPR \"Gornja\""
+	desc = "A modernized incendiary fresh out of serbia very flammable, for clearing out ship's and villages."
```
Comment (Firefox13) [inline code/modules/projectiles/ammunition/bullets.dm:503]:
```suggestion
	desc = "An incendiary grenade shell, intended for clearing out tight spaces such as ship interiors and village houses. With love from Serbia."
```

File: code/modules/projectiles/ammunition/bullets.dm [hunk 5]
```diff
@@ -474,6 +474,41 @@
...
+	w_class = ITEM_SIZE_SMALL
+	caliber = CAL_GRENADE
+	is_caseless = TRUE
+	maxamount = 1
+	spawn_tags = SPAWN_TAG_AMMO
+	rarity_value = 15
+
+/obj/item/ammo_casing/grenade/flash
+	name = "FS FBR \"Stallone\""
+	desc = "A Duty-grade Frozen Star flash round, for clearing out criminal hideouts."
```
Comment (Firefox13) [inline code/modules/projectiles/ammunition/bullets.dm:491]: What does "Duty-grade" mean here?

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 6]
```diff
@@ -299,6 +299,33 @@
...
+
+
+/obj/item/storage/box/whitephos_rounds
+	name = "box of white phosphorous rounds"
+	desc = "A box containing 6 White phosphorous rounds, designed to be fired from grenade launchers."
+	icon_state = "box_security"
+	illustration = "flashbang"
+	rarity_value = 60
+	prespawned_content_amount = 6
+	prespawned_content_type = /obj/item/ammo_casing/grenade/white_phosphorous
```
Comment (Firefox13) [inline code/game/objects/items/weapons/storage/boxes.dm:328]:
```suggestion
	prespawned_content_type = /obj/item/ammo_casing/grenade/white_phosphorous

```

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 7]
```diff
@@ -299,6 +299,33 @@
 	prespawned_content_amount = 6
 	prespawned_content_type = /obj/item/ammo_casing/grenade
 
+/obj/item/storage/box/flash_rounds
+	name = "box of Flash rounds"
+	desc = "A box containing 6 Flashbang rounds, designed to be fired from grenade launchers."
```
Comment (Firefox13) [inline code/game/objects/items/weapons/storage/boxes.dm:304]:
```suggestion
	desc = "A box containing 6 flashbang rounds, designed to be fired from grenade launchers."
```

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 8]
```diff
@@ -299,6 +299,33 @@
...
+	icon_state = "box_security"
+	illustration = "flashbang"
+	rarity_value = 60
+	prespawned_content_amount = 6
+	prespawned_content_type = /obj/item/ammo_casing/grenade/flash
+
+/obj/item/storage/box/smoke_rounds
+	name = "box of Smoke rounds"
```
Comment (Firefox13) [inline code/game/objects/items/weapons/storage/boxes.dm:312]:
```suggestion
	name = "box of smoke rounds"
```

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 9]
```diff
@@ -299,6 +299,33 @@
...
+	desc = "A box containing 6 Smoke rounds, designed to be fired from grenade launchers."
```
Comment (Firefox13) [inline code/game/objects/items/weapons/storage/boxes.dm:313]:
```suggestion
	desc = "A box containing 6 smoke rounds, designed to be fired from grenade launchers."
```

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 10]
```diff
@@ -299,6 +299,33 @@
...
+	icon_state = "box_security"
+	illustration = "flashbang"
+	rarity_value = 60
+	prespawned_content_amount = 6
+	prespawned_content_type = /obj/item/ammo_casing/grenade/smoke
+
+
+/obj/item/storage/box/whitephos_rounds
+	name = "box of white phosphorous rounds"
+	desc = "A box containing 6 White phosphorous rounds, designed to be fired from grenade launchers."
```
Comment (Firefox13) [inline code/game/objects/items/weapons/storage/boxes.dm:323]:
```suggestion
	desc = "A box containing 6 shite phosphorous rounds, designed to be fired from grenade launchers."
```

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 11]
```diff
@@ -299,6 +299,33 @@
 	prespawned_content_amount = 6
 	prespawned_content_type = /obj/item/ammo_casing/grenade
 
+/obj/item/storage/box/flash_rounds
+	name = "box of Flash rounds"
```
Comment (Firefox13) [inline code/game/objects/items/weapons/storage/boxes.dm:303]:
```suggestion
	name = "box of flash rounds"
```

**Review by Firefox13 — COMMENTED** [review]
Gotta figure out the smoke grenades still.

File: code/modules/projectiles/ammunition/bullets.dm (hunk 5 above)
Comment (thedarkcrusader) [reply code/modules/projectiles/ammunition/bullets.dm:491]: leftover i can just change it to standard issue  like the rest

**Comment by thedarkcrusader** [comment]
So close yet so so far away what a totally unfun place to be

**Comment by thedarkcrusader** [comment]
the more i rush this the more it goes downhill

**Comment by thedarkcrusader** [comment]
this is all good now

**Comment by thedarkcrusader** [comment]
im going to  assume that nothing here is broken for now

**Comment by thedarkcrusader** [comment]
@Firefox13  PR is ready for review  fixed smoke

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 12]
```diff
@@ -299,6 +299,24 @@
...
+	desc = "A box containing 6 flashbang rounds, designed to be fired from grenade launchers."
+	icon_state = "box_security"
+	illustration = "flashbang"
+	rarity_value = 60
+	prespawned_content_amount = 6
+	prespawned_content_type = /obj/item/ammo_casing/grenade/flash
+
+/obj/item/storage/box/whitephos_rounds
+	name = "box of white phosphorous rounds"
+	desc = "A box containing 6 shite phosphorous rounds, designed to be fired from grenade launchers."
```
Comment (TheShown911) [inline code/game/objects/items/weapons/storage/boxes.dm:313]: shite?

File: code/modules/projectiles/ammunition/bullets.dm [hunk 13]
```diff
@@ -474,6 +474,43 @@
...
+	w_class = ITEM_SIZE_SMALL
+	caliber = CAL_GRENADE
+	is_caseless = TRUE
+	maxamount = 1
+	spawn_tags = SPAWN_TAG_AMMO
+	rarity_value = 15
+
+
+/obj/item/ammo_casing/grenade/smoke
+	name = "FS FBR \"Stallone\""
```
Comment (TheShown911) [inline code/modules/projectiles/ammunition/bullets.dm:491]: These 2 got the same name

File: code/modules/projectiles/projectile/projectilegrenades.dm [hunk 14]
```diff
@@ -104,3 +104,26 @@
...
+	hand_gren = /obj/item/grenade/flashbang
+
+/obj/item/projectile/bullet/grenade/smoke
+	name = "smoke grenade"
+
+/obj/item/projectile/bullet/grenade/smoke/grenade_effect(target)
+	var/datum/effect/effect/system/smoke_spread/bad/smoke = new()
+	playsound(loc, 'sound/effects/smoke.ogg', 50, 1, -3)
+	// If this is >9 byond shits itself and crashes
+	smoke.set_up(10, 0, get_turf(target))
```
Comment (TheShown911) [inline code/modules/projectiles/projectile/projectilegrenades.dm:118]: 10 > 9

**Comment by thedarkcrusader** [comment]
one second

File: code/modules/projectiles/projectile/projectilegrenades.dm (hunk 14 above)
Comment (thedarkcrusader) [reply code/modules/projectiles/projectile/projectilegrenades.dm:118]: confused here  every other smoke setup has 10

**Review by TheShown911 — APPROVED** [review]
Gonna approve this, but waiting on fox to comment.

File: code/modules/projectiles/ammunition/bullets.dm [hunk 15]
```diff
@@ -474,6 +474,43 @@
...
+	caliber = CAL_GRENADE
+	is_caseless = TRUE
+	maxamount = 1
+	spawn_tags = SPAWN_TAG_AMMO
+	rarity_value = 15
+
+/obj/item/ammo_casing/grenade/white_phosphorous
+	name = "SA WPR \"Gornja\""
+	desc = "A modernized incendiary fresh out of serbia very flammable, for clearing out ship's and villages."
+	icon_state = "warcrime"
```
Comment (Firefox13) [inline code/modules/projectiles/ammunition/bullets.dm:505]: Change the name to something that represents the item.

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 16]
```diff
@@ -326,6 +344,16 @@
 	prespawned_content_amount = 6
 	prespawned_content_type =  /obj/item/ammo_casing/grenade/teargas
 
+/obj/item/storage/box/smoke
+	name = "box of smoke grenade Shells"
+	desc = "A box containing 6 smoke grenade shells for use with a launcher."
```
Comment (Firefox13) [inline code/game/objects/items/weapons/storage/boxes.dm:349]:
```suggestion
	desc = "A box containing 6 smoke grenade shells, designed to be fired from grenade launchers."
```
More in line with other boxes descriptions

File: code/game/objects/items/weapons/storage/boxes.dm (hunk 12 above)
Comment (Firefox13) [inline code/game/objects/items/weapons/storage/boxes.dm:313]:
```suggestion
	desc = "A box containing 6 white phosphorous rounds, designed to be fired from grenade launchers."
```

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 17]
```diff
@@ -326,6 +344,16 @@
 	prespawned_content_amount = 6
 	prespawned_content_type =  /obj/item/ammo_casing/grenade/teargas
 
+/obj/item/storage/box/smoke
+	name = "box of smoke grenade Shells"
```
Comment (Firefox13) [inline code/game/objects/items/weapons/storage/boxes.dm:348]:
```suggestion
	name = "box of smoke grenade rounds"
```

**Comment by thedarkcrusader** [comment]
-Re tested every grenade

-Spawned every grenade type and box
-  boxes and correct ammo types spawned inside them
- loaded every ammo type inside of a launcher every ammo type did as intend

**Review by Firefox13 — APPROVED** [review]

**Pull request merged**